src/components/users/UserLoginForm.js

```diff
@@ -0,0 +1,91 @@
+import React, { Component } from 'react'
+import emailValidator from 'email-validator'
+
+import RaisedButton from 'material-ui/RaisedButton'
+
+import TextField from './TextFieldExtended'
+
+class UserLoginForm extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      password: null,
+      email: null,
+
+      validFields: {
+        password: false,
+        email: false
+      }
+    }
+  }
+
+  onFieldValueChange = (fieldName, fieldValue, hasError) => {
+    let newState = Object.assign({}, this.state)
```

Comment:
You don't need to preserve old state to update it. When calling setState(), React merges that object with current state.